.github/actions/composite/updateProtectedBranch/action.yml

```diff
@@ -134,12 +124,24 @@ runs:
         GITHUB_TOKEN: ${{ inputs.OS_BOTIFY_TOKEN }}
 
     - name: Fail workflow if PR is not mergeable
-      if: ${{ steps.isPullRequestMergeable.outputs.IS_MERGEABLE == 'false' }}
+      if: ${{ inputs.TARGET_BRANCH == 'main' && steps.isPullRequestMergeable.outputs.IS_MERGEABLE == 'false' }}
       shell: bash
       run: exit 1
 
     - name: Auto-merge the PR
+      if: ${{ inputs.TARGET_BRANCH == 'main' }}
       shell: bash
       run: gh pr merge ${{ steps.createPullRequest.outputs.PR_NUMBER }} --merge --delete-branch
       env:
         GITHUB_TOKEN: ${{ inputs.OS_BOTIFY_TOKEN }}
+
+    - name: Delete staging/production and replace it with contents of main/staging
+      if: ${{ inputs.TARGET_BRANCH != 'main' }}
+      shell: bash
+      run: |
+        git checkout ${{ env.SOURCE_BRANCH }}
```

Comment:
Do we want to also verify that `SOURCE_BRANCH` is either `main` or `staging` first? Based on line 47, it could be something else too (though I don't understand why)

Reply:
So the `Set source branch` step sets the source branch in the environment like so:

1. If the target branch is `staging`, the source branch is `main`
1. If the target branch is `production`, the source branch is `staging`
1. Else (the target branch is `main`), and the source branch is whatever is passed in `inputs.TARGET_BRANCH`

Reply:
Ok, so this is covered by the `if: ${{ inputs.TARGET_BRANCH != 'main' }}`. That's all I wanted to confirm, thanks.

Reply:
Yeah. Overall this action maybe isn't as clear and clean as it could be because so much of it only applies for if the target branch is main, but cleanup is out-of-scope for now. I'll be working on a design doc soon to clean up a lot of our GitHub Actions code.